neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -215,6 +221,46 @@ func (g Guest) ValidateForMemoryProvider(p MemoryProvider) error {
 	return nil
 }
 
+// LogicalTime allows to track progress of changes to a VM.
+type LogicalTime struct {
+	Value     int64       `json:"value"`
+	UpdatedAt metav1.Time `json:"updatedAt"`
+}
+
+// MarshalLogObject implements zapcore.ObjectMarshaler, so that LogicalTime can be used with zap.Object
+func (r *LogicalTime) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddInt64("value", r.Value)
+	enc.AddTime("updatedAt", r.UpdatedAt.Time)
+	return nil
+}
+
+func (t *LogicalTime) Rewind(now time.Time) *LogicalTime {
+	if t == nil {
+		return nil
+	}
+	return &LogicalTime{
+		Value:     t.Value,
+		UpdatedAt: metav1.NewTime(now),
+	}
+}
```

Comment:
What does `Rewind` do? Without context, the name of the function does not give me any clues. (maybe it should be called e.g. `OverrideRealTime` or something?)